### PR TITLE
did the task

### PR DESCRIPTION
**Pull request by vanyachyzh**
[DEMO LINK](https://vanyachyzh.github.io/react_carousel/)

### CODE REVIEW

File: src/components/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+class Carousel extends Component<Props, State> {
+  state: State = {
+    itemWidth: this.props.itemWidth,
+    frameSize: this.props.frameSize,
+    step: this.props.step,
+    animationDuration: this.props.animationDuration,
+    coordX: 0,
+  };
+
+  swipe = (direction: string) => {
```
Comment (OlEzhA0) [inline src/components/Carousel/Carousel.tsx:31]: would be better to type it like
```suggestion
  swipe = (direction: 'prev' | 'next') => {
```

File: src/components/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+        <input
+          type="text"
+          placeholder="Step"
+          className="Carousel__input"
+          onChange={(e) => {
+            this.setState({ step: +e.target.value });
+          }}
+        />
+
```
Comment (OlEzhA0) [inline src/components/Carousel/Carousel.tsx:148]: where is checkbox for isFinite property ?)

File: src/components/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+              transform: `translateX(${coordX > 0 ? -coordX : coordX}px)`,
+              transition: `transform ${animationDuration}ms ease-in-out`,
+            }}
+          >
+            {images.map(image => (
+              <li key={image}>
+                <img
+                  src={image}
+                  alt="1"
+                  style={{ width: `${itemWidth}px` }}
```
Comment (OlEzhA0) [inline src/components/Carousel/Carousel.tsx:92]: you can pass just number value
```suggestion
                  style={{ width: itemWidth }}
```

**Review by OlEzhA0 — CHANGES_REQUESTED** [review]
hey! please add isFinite checkbox to demo)

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]
- Here should be shown default values 
image

- Would be great to add a min value for all input. See some examples below

https://user-images.githubusercontent.com/99403198/222414890-64b58d2a-6a05-42ae-afa6-a021150f3f73.mov

https://user-images.githubusercontent.com/99403198/222415224-d5b68293-e19d-4191-b760-b82d7d9024e0.mov

- You should disable buttons if the user can't scroll anymore 

https://user-images.githubusercontent.com/99403198/222415534-28d0d73e-0fc1-48f4-9cad-06b0b20f9376.mov

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**